### PR TITLE
tests for RecipeEntity now fixed. Ready for review.

### PR DESCRIPTION
**Pull request by cwoodwarduk**
Test fixes ready for review.

### CODE REVIEW

File: tests/src/Recipe/RecipeEntity.php [hunk 1]
```diff
@@ -74,18 +74,16 @@ public function testMalformedRecipeEntityGenerateHTML()
...
+        $result = $testIngredient->GenerateHTMLSaved();
         $expected = "<div class='recipe'>";
         $expected .= "<div class='left'>";
         $expected .= "<div class='recipeImage'>";
         $expected .= "<img src='string'/>";
         $expected .= "</div>";
-        $expected .= "<a href='string'>";
-        $expected .= "<button>Link to recipe</button>";
-        $expected .= "</a>";
+        $expected .= "<a name='deleteRecipe'><button>Delete Recipe</button></a>";
```
Comment (jdselby24) [inline tests/src/Recipe/RecipeEntity.php]: Needs hidden form, see marcs addition to recipeEntitiy

File: tests/src/Recipe/RecipeEntity.php [hunk 2]
```diff
@@ -94,24 +92,21 @@ public function testSuccessRecipeEntity()
...
+        $result = $testIngredient->GenerateHTMLSaved();
         $expected = "<div class='recipe'>";
         $expected .= "<div class='left'>";
         $expected .= "<div class='recipeImage'>";
         $expected .= "<img src='9'/>";
         $expected .= "</div>";
-        $expected .= "<a href='9'>";
-        $expected .= "<button>Link to recipe</button>";
-        $expected .= "</a>";
+        $expected .= "<a name='deleteRecipe'><button>Delete Recipe</button></a>";
```
Comment (jdselby24) [inline tests/src/Recipe/RecipeEntity.php]: Needs hidden form, see marcs addition to recipeEntitiy

**Review by jdselby24 — CHANGES_REQUESTED** [review]
Needs hidden form, see marcs addition to recipeEntitiy

File: tests/src/Recipe/RecipeEntity.php (hunk 1 above)
Comment (cwoodwarduk) [reply tests/src/Recipe/RecipeEntity.php]: Fixed

File: tests/src/Recipe/RecipeEntity.php (hunk 2 above)
Comment (cwoodwarduk) [reply tests/src/Recipe/RecipeEntity.php]: Fixed

**Review by jdselby24 — CHANGES_REQUESTED** [review]
Change button name to delete rather than save

File: src/Recipe/RecipeEntity.php [hunk 3]
```diff
@@ -73,7 +73,16 @@ public function generateHTML(): string
         $html .= "<div class='recipeImage'>";
         $html .= "<img src='$this->imageURL'/>";
         $html .= "</div>";
-        $html .= "<a href='$this->link'><button>Link to recipe</button></a>";
+        $html .= "<form class='recipeForm'>";
+        $html .= "<input class='hidden' type='text' name='title' value='$this->title'>";
+        $html .= "<input class='hidden' type='text' name='ingredients' value='$this->ingredients'>";
+        $html .= "<input class='hidden' type='text' name='imageURL' value='$this->imageURL'>";
+        $html .= "<input class='hidden' type='text' name='recipeLink' value='$this->link'>";
```
Comment (jdselby24) [inline src/Recipe/RecipeEntity.php]:
```suggestion
        $html .= "<input class='hidden' type='text' name='link' value='$this->link'>";
```

File: src/Recipe/RecipeEntity.php [hunk 4]
```diff
@@ -95,8 +104,17 @@ public function generateHTMLSaved(): string
         $html .= "<div class='recipeImage'>";
         $html .= "<img src='$this->imageURL'/>";
         $html .= "</div>";
+        $html .= "<form class='recipeForm'>";
+        $html .= "<input class='hidden' type='text' name='title' value='$this->title'>";
+        $html .= "<input class='hidden' type='text' name='ingredients' value='$this->ingredients'>";
+        $html .= "<input class='hidden' type='text' name='imageURL' value='$this->imageURL'>";
+        $html .= "<input class='hidden' type='text' name='recipeLink' value='$this->link'>";
```
Comment (jdselby24) [inline src/Recipe/RecipeEntity.php]:
```suggestion
        $html .= "<input class='hidden' type='text' name='link' value='$this->link'>";
```

File: tests/src/Recipe/RecipeEntity.php [hunk 5]
```diff
@@ -20,9 +20,16 @@ public function testSuccessRecipeEntityGenerateHTML()
...
         $expected .= "<img src='string'/>";
         $expected .= "</div>";
-        $expected .= "<a href='string'>";
-        $expected .= "<button>Link to recipe</button>";
-        $expected .= "</a>";
+        $expected .= "<form class='recipeForm'>";
+        $expected .= "<input class='hidden' type='text' name='title' value='string'>";
+        $expected .= "<input class='hidden' type='text' name='ingredients' value='string'>";
+        $expected .= "<input class='hidden' type='text' name='imageURL' value='string'>";
+        $expected .= "<input class='hidden' type='text' name='recipeLink' value='string'>";
```
Comment (jdselby24) [inline tests/src/Recipe/RecipeEntity.php]:
```suggestion
        $html .= "<input class='hidden' type='text' name='link' value='$this->link'>";
```

File: tests/src/Recipe/RecipeEntity.php [hunk 6]
```diff
@@ -46,9 +53,16 @@ public function testFailureRecipeEntityGenerateHTML()
...
         $expected .= "<img src='9'/>";
         $expected .= "</div>";
-        $expected .= "<a href='9'>";
-        $expected .= "<button>Link to recipe</button>";
-        $expected .= "</a>";
+        $expected .= "<form class='recipeForm'>";
+        $expected .= "<input class='hidden' type='text' name='title' value='9'>";
+        $expected .= "<input class='hidden' type='text' name='ingredients' value='9'>";
+        $expected .= "<input class='hidden' type='text' name='imageURL' value='9'>";
+        $expected .= "<input class='hidden' type='text' name='recipeLink' value='9'>";
```
Comment (jdselby24) [inline tests/src/Recipe/RecipeEntity.php]:
```suggestion
        $html .= "<input class='hidden' type='text' name='link' value='$this->link'>";
```

File: tests/src/Recipe/RecipeEntity.php [hunk 7]
```diff
@@ -74,18 +88,25 @@ public function testMalformedRecipeEntityGenerateHTML()
...
         $expected .= "<img src='string'/>";
         $expected .= "</div>";
-        $expected .= "<a href='string'>";
-        $expected .= "<button>Link to recipe</button>";
-        $expected .= "</a>";
+        $expected .= "<form class='recipeForm'>";
+        $expected .= "<input class='hidden' type='text' name='title' value='string'>";
+        $expected .= "<input class='hidden' type='text' name='ingredients' value='string'>";
+        $expected .= "<input class='hidden' type='text' name='imageURL' value='string'>";
+        $expected .= "<input class='hidden' type='text' name='recipeLink' value='string'>";
```
Comment (jdselby24) [inline tests/src/Recipe/RecipeEntity.php]:
```suggestion
        $html .= "<input class='hidden' type='text' name='link' value='$this->link'>";
```

File: tests/src/Recipe/RecipeEntity.php [hunk 8]
```diff
@@ -94,24 +115,30 @@ public function testSuccessRecipeEntity()
...
         $expected .= "<img src='9'/>";
         $expected .= "</div>";
-        $expected .= "<a href='9'>";
-        $expected .= "<button>Link to recipe</button>";
-        $expected .= "</a>";
+        $expected .= "<form class='recipeForm'>";
+        $expected .= "<input class='hidden' type='text' name='title' value='9'>";
+        $expected .= "<input class='hidden' type='text' name='ingredients' value='9'>";
+        $expected .= "<input class='hidden' type='text' name='imageURL' value='9'>";
+        $expected .= "<input class='hidden' type='text' name='recipeLink' value='9'>";
```
Comment (jdselby24) [inline tests/src/Recipe/RecipeEntity.php]:
```suggestion
        $html .= "<input class='hidden' type='text' name='link' value='$this->link'>";
```

**Review by jdselby24 — CHANGES_REQUESTED** [review]
```suggestion
        $html .= "<input class='hidden' type='text' name='link' value='$this->link'>";
```

**Review by jdselby24 — APPROVED** [review]

**Pull request merged**